pipes/WDL/workflows/scaffold_and_refine_multitaxa.wdl

```diff
@@ -0,0 +1,112 @@
+
+workflow scaffold_and_refine_multitaxa {
+    meta {
+        description: "Scaffold de novo contigs against a set of possible references and subsequently polish with reads."
+        author: "Broad Viral Genomics"
+        email:  "viral-ngs@broadinstitute.org"
+        allowNestedInputs: true
+    }
+
+    input {
+        String  sample_id
+        File    reads_unmapped_bam
+
+        Array[Pair[Int,Array[String]+]] taxid_to_ref_accessions = [
+            (208893,  ["KY654518.1"]),  # RSV-A
+            (208895,  ["MZ516105.1"]),  # RSV-B
+            (573824,  ["NC_038311.1"]), # Rhino A1
+            (147711,  ["NC_001617.1"]), # Rhino A2
+            (185900,  ["ON311191.1"]),  # Rhino B27
+            (1418033, ["ON311169.1"]),  # Rhino C15
+            (2697049, ["NC_045512.2"]), # SARS-CoV-2 Wuhan Hu-1
+            (641809,  ["NC_026438.1", "NC_026435.1", "NC_026437.1", "NC_026433.1", "NC_026436.1", "NC_026434.1", "NC_026431.1", "NC_026432.1"]),  # Flu A/California/07/2009 H1N1
+            (335341,  ["NC_007373.1", "NC_007372.1", "NC_007371.1", "NC_007366.1", "NC_007369.1", "NC_007368.1", "NC_007367.1", "NC_007370.1"]),  # Flu A/New York/392/2004 H3N2
+            (518987,  ["NC_002204.1", "NC_002205.1", "NC_002206.1", "NC_002207.1", "NC_002208.1", "NC_002209.1", "NC_002210.1", "NC_002211.1"]),  # Flu B/Lee/1940
+            (162145,  ["NC_039199.1"]), # metapneumo
+            (12730,   ["NC_003461.1"]), # paraflu 1
+            (11216,   ["NC_001796.2"])  # paraflu 3
+        ]
+    }
+
+    Array[String] assembly_header = ["sample_id", "taxid", "assembly_fasta", "aligned_only_reads_bam", "coverage_plot", "assembly_length", "assembly_length_unambiguous", "reads_aligned", "mean_coverage", "intermediate_gapfill_fasta", "assembly_preimpute_length_unambiguous", "replicate_concordant_sites", "replicate_discordant_snps", "replicate_discordant_indels", "replicate_discordant_vcf", "isnvsFile", "aligned_bam", "coverage_tsv", "read_pairs_aligned", "bases_aligned"]
+
+    scatter(taxon in taxid_to_ref_accessions) {
+        call ncbi.download_annotations {
+            input:
+                accessions = taxon.right,
+                combined_out_prefix = taxon.left
+        }
+        call assembly.scaffold {
+            input:
+                reads_bam = reads_unmapped_bam,
+                reference_genome_fasta = [download_annotations.combined_fasta],
+                min_length_fraction = 0,
+                min_unambig = 0,
+                allow_incomplete_output = true
+        }
+        call assemble_refbased.assemble_refbased as refine {
+            input:
+                reads_unmapped_bams = [reads_unmapped_bam],
+                reference_fasta     = scaffold.scaffold_fasta,
+                sample_name         = sample_id
+        }
+        # to do: if pre-impute unambig length > some fraction of ref genome, run ncbi.rename_fasta_header and ncbi.align_and_annot_transfer_single
```

Comment:
`some fraction of ref genome` will be a workflow parameter?

Reply:
Also, maybe we try [Liftoff](https://github.com/agshumate/Liftoff) for annotation transfer (no chain file required—it does the alignment via minimap2).

Reply:
Yeah I think that's right... this is all obviously to do at some future stage.